crates/electrum/tests/test_electrum.rs

```diff
@@ -559,3 +559,116 @@ fn test_sync_with_coinbase() -> anyhow::Result<()> {
+
+    // Setup receiver.
+    let (mut recv_chain, _) = LocalChain::from_genesis_hash(env.bitcoind.client.get_block_hash(0)?);
+    let mut recv_graph = IndexedTxGraph::<ConfirmationBlockTime, _>::new({
+        let mut recv_index = SpkTxOutIndex::default();
+        recv_index.insert_spk((), spk_to_track.clone());
+        recv_index
+    });
+
+    // Mine some blocks.
+    env.mine_blocks(101, None)?;
+
+    // Send a preliminary tx such that the new utxo in Core's wallet
+    // becomes the input of the next tx
+    let new_addr = env
+        .rpc_client()
+        .get_new_address(None, None)?
+        .assume_checked();
+    let prev_amt = SEND_AMOUNT + Amount::from_sat(1650);
+    env.send(&new_addr, prev_amt)?;
+    let prev_hash_block = env.mine_blocks(1, None)?.into_iter().next();
+
+    let txid = env.send(&addr_to_track, SEND_AMOUNT)?;
+
+    // Mine a block to confirm sent tx.
+    let hash_block = env.mine_blocks(1, None)?.into_iter().next();
+
+    // Look at the tx we just sent, it should have 1 input and 1 output
+    let tx = env
+        .rpc_client()
+        .get_raw_transaction_info(&txid, hash_block.as_ref())?;
+    assert_eq!(tx.vin.len(), 1);
+    assert_eq!(tx.vout.len(), 1);
+    let vin = &tx.vin[0];
+    let prev_txid = vin.txid.unwrap();
+    let vout = vin.vout.unwrap();
+    let outpoint = bdk_chain::bitcoin::OutPoint::new(prev_txid, vout);
+
+    // Get the txout of the previous tx
+    let prev_tx = env
+        .rpc_client()
+        .get_raw_transaction_info(&prev_txid, prev_hash_block.as_ref())?;
+    let txout = prev_tx
+        .vout
+        .iter()
+        .find(|txout| txout.value == prev_amt)
+        .unwrap();
+    let script_pubkey = ScriptBuf::from_bytes(txout.script_pub_key.hex.to_vec());
+    let txout = bdk_chain::bitcoin::TxOut {
+        value: txout.value,
+        script_pubkey,
+    };
+
+    // Sync up to tip.
+    env.wait_until_electrum_sees_block(Duration::from_secs(6))?;
+    let _ = sync_with_electrum(
+        &client,
+        [spk_to_track.clone()],
+        &mut recv_chain,
+        &mut recv_graph,
+    )?;
+
+    // Check the graph update contains the right floating txout
+    let graph_txout = recv_graph
+        .graph()
+        .all_txouts()
+        .find(|(_op, txout)| txout.value == prev_amt)
+        .unwrap();
+    assert_eq!(graph_txout, (outpoint, &txout));
+
+    // Check to see if tx is confirmed.
+    assert_eq!(
+        get_balance(&recv_chain, &recv_graph)?,
+        Balance {
+            confirmed: SEND_AMOUNT,
+            ..Balance::default()
+        },
+    );
+
+    for tx in recv_graph.graph().full_txs() {
+        // Retrieve the calculated fee from `TxGraph`, which will panic if we do not have the
+        // floating txouts available from the transaction's previous outputs.
+        let fee = recv_graph
+            .graph()
+            .calculate_fee(&tx.tx)
+            .expect("fee must exist");
+
```

Comment:
❓ You could probably assert the expected fee here too, to ensure it's the same 1650 as expected. WDYT?

Reply:
Also, maybe use it as constant like SEND_AMOUNT, e.g FEE_AMOUNT.

Reply:
good point @oleonardolima I am going to add the additional assert.